pkg/scheduler/framework/plugins/imagelocality/image_locality.go

```diff
@@ -63,7 +63,11 @@ func (pl *ImageLocality) Score(ctx context.Context, state *framework.CycleState,
 	}
 	totalNumNodes := len(nodeInfos)
 
-	score := calculatePriority(sumImageScores(nodeInfo, pod.Spec.Containers, totalNumNodes), len(pod.Spec.Containers))
+	imageScores := sumImageScores(nodeInfo, pod.Spec.Containers, totalNumNodes)
```

Comment:
I think it's better to append `pod.Spec.Containers` and `pod.Spec.InitContainers` together as a single slice and call `sumImageScores` once.

Reply:
This will use more memory, and there is no difference in execution efficiency. WDYT?

Reply:
And you need to check the length of `InitContainers` before you call `sumImageScores` a second time.

> I think it's better to append `pod.Spec.Containers` and `pod.Spec.InitContainers` together as a single slice and call sumImageScores once.

Sorry. Maybe I didn't express my idea clearly.  My suggestion is that since `InitContainer` and `Containers` are started sequentially, we need to consider them uniformly when calculating the score, e.g. if it already exists in `initContainer`, the image should not be calculated in `Containers` again.

Reply:
Add a set[string], save all `Containers` and `InitContainers`, and then call sumImageScores once to calculate the score. Would this be more reasonable？

Reply:
> Add a set[string], save all Containers and InitContainers, and then call sumImageScores once to calculate the score.

This is still a double calculation of the same images in `Containers` and `InitContainers`. You can input `Containers` and `InitContainers` as two different arguments in the func `sumImageScores` and use a `set[string]` to avoid double calculation.